flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -699,6 +701,32 @@ public void testDisableSystemClassPathIncludeUserJarAndWithIllegalShipDirectoryN
         }
     }
 
+    /** Tests that the usrlib will be automatically shipped. */
+    @Test
+    public void testShipUsrLib() throws IOException {
+        final Map<String, String> oldEnv = System.getenv();
+        final Map<String, String> env = new HashMap<>(1);
+        File homeFolder = temporaryFolder.newFolder().getAbsoluteFile();
+        File libFolder = new File(homeFolder.getAbsolutePath(), "lib");
+        assertTrue(libFolder.createNewFile());
+        File usrLibFolder =
+                new File(homeFolder.getAbsolutePath(), ConfigConstants.DEFAULT_FLINK_USR_LIB_DIR);
+        assertTrue(usrLibFolder.mkdirs());
+        File usrLibFile = new File(usrLibFolder, "usrLibFile.jar");
+        assertTrue(usrLibFile.createNewFile());
+        env.put(ConfigConstants.ENV_FLINK_LIB_DIR, libFolder.getAbsolutePath());
+        CommonTestUtils.setEnv(env);
+
+        try (YarnClusterDescriptor descriptor = createYarnClusterDescriptor()) {
+            Set<File> effectiveShipFiles = new HashSet<>();
+            descriptor.addUsrLibFolderToShipFiles(effectiveShipFiles);
+            assertThat(effectiveShipFiles, containsInAnyOrder(usrLibFolder));
+            assertThat(effectiveShipFiles, not(containsInAnyOrder(usrLibFile)));
```

Comment:
I do not think we still need this assert. The above `containsInAnyOrder` could cover such case.

Reply:
The `assertThat(effectiveShipFiles, not(containsInAnyOrder(usrLibFile)));` just follows the check logic in testEnvironmentDirectoryShipping to make sure single is not added.

Reply:
You are right. Our `tryFindUserLibDirectory` will only return a directory.